### PR TITLE
Add sig labels and use aliases in OWNERS files

### PR DESCRIPTION
**Pull request by cblecker**
This PR does the following:
- Creates a sig-*-leads alias for all sigs and working groups
- Replaces the reviewers/approvers in the sig folders and design-proposals folders with that alias
- Adds a label field to the OWNERS files to allow for automatic sig labelling (https://github.com/kubernetes/test-infra/pull/3502)

### CODE REVIEW

File: contributors/design-proposals/cloud-provider/OWNERS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+reviewers:
+  - sig-cloud-provider-leads
+approvers:
+  - sig-cloud-provider-leads
+labels:
+  - sig/cloud-provider
```
Comment (xiangpengzhao) [inline contributors/design-proposals/cloud-provider/OWNERS]: We don't have this label.

File: sig-on-premise/OWNERS [hunk 2]
```diff
@@ -1,8 +1,6 @@
...
-  - marcoceppi
-  - dghubble
+  - sig-on-premise-leads
 approvers:
-  - zen
-  - marcoceppi
-  - dghubble
+  - sig-on-premise-leads
+labels:
+  - sig/onprem
```
Comment (xiangpengzhao) [inline sig-on-premise/OWNERS]: Hope the names can be consistent one day... onprem vs on-premise
ref: https://github.com/kubernetes/community/issues/703

File: sig-product-management/OWNERS [hunk 3]
```diff
@@ -1,8 +1,6 @@
...
-  - idvoretskyi
-  - calebamiles
+  - sig-product-management-leads
 approvers:
-  - apsinha
-  - idvoretskyi
-  - calebamiles
+  - sig-product-management-leads
+labels:
+  - sig/pm
```
Comment (xiangpengzhao) [inline sig-product-management/OWNERS]: pm vs product-management

File: contributors/design-proposals/cloud-provider/OWNERS (hunk 1 above)
Comment (cblecker) [reply contributors/design-proposals/cloud-provider/OWNERS]: Good catch

File: sig-on-premise/OWNERS (hunk 2 above)
Comment (cblecker) [reply sig-on-premise/OWNERS]: I hope they can to. This isn't to change that though -- just capturing the current state of the world.

**Comment by grodrigues3** [comment]
I like the idea in theory, but I'm concerned that this conflates sig-lead with technical lead.  Sig leads are responsible for organizing and driving the sig and if I recall correctly, there was an explicit decision to keep sig leads separate from tech leads.

That said, it seems to make sense for most of the design proposals dirs as it's a direct replacement of names with alias.

**Comment by cblecker** [comment]
@grodrigues3 I figured this was a good starting place. Most of the `contributors/design-proposals/*` directories didn't have an owners file to begin with, as those directories were created by you in #1010. The sig leads would be welcome to add to the OWNERS files and delegate as needed. The other major plus of this would be auto-labelling PRs based on the files they touch.

**Comment by grodrigues3** [comment]
This sounds good to me.  I'll tag contribX and see if we get any additional feedback.  otherwise, i'll LGTM this afternoon.

@kubernetes/sig-contributor-experience-pr-reviews

**Pull request merged**